core/iwasm/common/wasm_shared_memory.c

```diff
@@ -380,11 +384,23 @@ static bool
 map_remove_wait_info(HashMap *wait_map_, AtomicWaitInfo *wait_info,
                      void *address)
 {
+    os_mutex_lock(&wait_map_lock);
+    --wait_info->count_acquisition;
```

Comment:
what's the difference between using `count_acquisition` and `wait_info->wait_list->len` directly?

Reply:
Because `wait_info->wait_list` is empty when we acquire and it causes the problem, but I can try to reorder it so we can use `wait_info->wait_list->len directly`